diff --git a/node_modules/detox/android/detox/build.gradle b/node_modules/detox/android/detox/build.gradle
index 1e7358b..ac405e0 100644
--- a/node_modules/detox/android/detox/build.gradle
+++ b/node_modules/detox/android/detox/build.gradle
@@ -19,10 +19,10 @@ android {
 
     productFlavors {
         flavorDimensions "minReactNative"
-        minReactNative44 {
-            dimension "minReactNative"
+        //minReactNative44 {
+        //    dimension "minReactNative"
 
-        }
+        //}
         minReactNative46 {
             dimension "minReactNative"
         }
@@ -61,14 +61,19 @@ android {
     lintOptions {
         abortOnError false
     }
+
+    configurations.all {
+        resolutionStrategy.force 'com.google.code.findbugs:jsr305:3.0.2'
+        resolutionStrategy.force 'com.android.support:support-annotations:28.0.0'
+    }
 }
 
 dependencies {
-    minReactNative44Implementation 'com.squareup.okhttp3:okhttp:3.4.1'
-    minReactNative44Implementation 'com.squareup.okhttp3:okhttp-ws:3.4.1'
+    //minReactNative44Implementation 'com.squareup.okhttp3:okhttp:3.4.1'
+    //minReactNative44Implementation 'com.squareup.okhttp3:okhttp-ws:3.4.1'
 
-    minReactNative46Implementation 'com.squareup.okhttp3:okhttp:3.6.0'
-    minReactNative46Implementation 'com.squareup.okio:okio:1.13.0'
+    minReactNative46Implementation 'com.squareup.okhttp3:okhttp:3.12.1'
+    minReactNative46Implementation 'com.squareup.okio:okio:1.15.0'
 
     implementation('com.android.support.test.espresso:espresso-core:3.0.2', {
         exclude group: 'com.google.code.findbugs'
