patch-package
--- a/node_modules/detox/android/detox/build.gradle
+++ b/node_modules/detox/android/detox/build.gradle
@@ -22,10 +22,6 @@

     productFlavors {
         flavorDimensions "minReactNative"
-        minReactNative44 {
-            dimension "minReactNative"
-
-        }
         minReactNative46 {
             dimension "minReactNative"
         }
@@ -69,8 +65,6 @@
 dependencies {
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$_kotlinVersion"

-    minReactNative44Implementation 'com.squareup.okhttp3:okhttp:3.4.1'
-    minReactNative44Implementation 'com.squareup.okhttp3:okhttp-ws:3.4.1'

     minReactNative46Implementation 'com.squareup.okhttp3:okhttp:3.6.0'
     minReactNative46Implementation 'com.squareup.okio:okio:1.13.0'
@@ -82,7 +76,7 @@
     implementation 'com.android.support.test:rules:1.0.2'

     // noinspection GradleDynamicVersion
-    compileOnly "com.facebook.react:react-native:+"
+    api "com.facebook.react:react-native:+"

     implementation 'org.apache.commons:commons-lang3:3.4'
     implementation 'com.android.support.test.uiautomator:uiautomator-v18:2.1.3'
