diff --git a/node_modules/detox/android/detox/src/main/java/com/wix/detox/reactnative/idlingresources/UIModuleIdlingResource.java b/node_modules/detox/android/detox/src/main/java/com/wix/detox/reactnative/idlingresources/UIModuleIdlingResource.java
index 181233a..f834f6b 100644
--- a/node_modules/detox/android/detox/src/main/java/com/wix/detox/reactnative/idlingresources/UIModuleIdlingResource.java
+++ b/node_modules/detox/android/detox/src/main/java/com/wix/detox/reactnative/idlingresources/UIModuleIdlingResource.java
@@ -100,18 +100,21 @@ public class UIModuleIdlingResource implements IdlingResource, Choreographer.Fra
             }
 
             boolean isOperationQueueEmpty = (Boolean) Reflect.on(uiOperationQueue).call(METHOD_IS_EMPTY).get();
+            Log.d(LOG_TAG, "UIManagerModule queue status: runnables? / nonBatchesOps? / operationQueue? - " + runnablesAreEmpty + " / " + nonBatchesOpsEmpty + " / " + isOperationQueueEmpty);
 
             if (runnablesAreEmpty && nonBatchesOpsEmpty && isOperationQueueEmpty) {
                 if (callback != null) {
                     callback.onTransitionToIdle();
                 }
-                // Log.i(LOG_TAG, "UIManagerModule is idle.");
+                Log.d(LOG_TAG, "UIManagerModule is idle.");
                 return true;
             }
 
             Log.i(LOG_TAG, "UIManagerModule is busy.");
-            Choreographer.getInstance().postFrameCallback(this);
-            return false;
+            Log.w(LOG_TAG, "UIManagerModule is busy but damn the torpedoes!");
+            return true;
+            // Choreographer.getInstance().postFrameCallback(this);
+            // return false;
         } catch (ReflectException e) {
             Log.e(LOG_TAG, "Can't set up RN UIModule listener", e.getCause());
         }
