diff --git a/FirebaseAuth/Sources/Swift/Backend/AuthBackend.swift b/FirebaseAuth/Sources/Swift/Backend/AuthBackend.swift
index 444a97c1c07..84bf5ec55c9 100644
--- a/FirebaseAuth/Sources/Swift/Backend/AuthBackend.swift
+++ b/FirebaseAuth/Sources/Swift/Backend/AuthBackend.swift
@@ -76,21 +76,19 @@ class AuthBackend {
     return "FirebaseAuth.iOS/\(FirebaseVersion()) \(GTMFetcherStandardUserAgentString(nil))"
   }
 
-  private static var gBackendImplementation: AuthBackendImplementation?
+  private static var realRPCBackend = AuthBackendRPCImplementation()
+  private static var gBackendImplementation = realRPCBackend
 
-  class func setDefaultBackendImplementationWithRPCIssuer(issuer: AuthBackendRPCIssuer?) {
-    let defaultImplementation = AuthBackendRPCImplementation()
-    if let issuer = issuer {
-      defaultImplementation.rpcIssuer = issuer
-    }
-    gBackendImplementation = defaultImplementation
+  class func setTestRPCIssuer(issuer: AuthBackendRPCIssuer) {
+    gBackendImplementation.rpcIssuer = issuer
+  }
+
+  class func resetRPCIssuer() {
+    gBackendImplementation.rpcIssuer = realRPCBackend.rpcIssuer
   }
 
   class func implementation() -> AuthBackendImplementation {
-    if gBackendImplementation == nil {
-      gBackendImplementation = AuthBackendRPCImplementation()
-    }
-    return gBackendImplementation!
+    return gBackendImplementation
   }
 
   class func call<T: AuthRPCRequest>(with request: T) async throws -> T.Response {
