diff --git a/node_modules/jet/lib/node/patch/detox.js b/node_modules/jet/lib/node/patch/detox.js
index fdca559..5c74b90 100644
--- a/node_modules/jet/lib/node/patch/detox.js
+++ b/node_modules/jet/lib/node/patch/detox.js
@@ -28,6 +28,9 @@ if (detox) {
       // device.reloadReactNative({ ... })
       // todo detoxOriginalReloadReactNative currently broken it seems
       // const detoxOriginalReloadReactNative = originalDevice.reloadReactNative.bind(originalDevice);
+      if (!originalDevice) {
+        return undefined;
+      }
       originalDevice.reloadReactNative = async () => {
         ready.reset();
         global.jet.reload();
@@ -58,13 +61,6 @@ if (detox) {
    *   DETOX OVERRIDES
    * ------------------- */
 
-  // detox.init()
-  const detoxOriginalInit = detox.init.bind(detox);
-  detox.init = async (...args) => {
-    ready.reset();
-    await detoxOriginalInit(...args);
-    return ready.wait();
-  };
   // TODO: Salakar: all of this will go in first release
   // TODO: Salakar: jet global will provide it's own methods that internally call these
 
