diff --git a/node_modules/react-native-macos/Libraries/Blob/BlobManager.js b/node_modules/react-native-macos/Libraries/Blob/BlobManager.js
index f4c5ed374..f7afaf4e9 100644
--- a/node_modules/react-native-macos/Libraries/Blob/BlobManager.js
+++ b/node_modules/react-native-macos/Libraries/Blob/BlobManager.js
@@ -9,7 +9,7 @@
  */
 
 import type {BlobCollector, BlobData, BlobOptions} from './BlobTypes';
-
+import {fromByteArray} from 'base64-js';
 import NativeBlobModule from './NativeBlobModule';
 import invariant from 'invariant';
 
@@ -67,9 +67,11 @@
     const blobId = uuidv4();
     const items = parts.map(part => {
       if (part instanceof ArrayBuffer || ArrayBuffer.isView(part)) {
-        throw new Error(
-          "Creating blobs from 'ArrayBuffer' and 'ArrayBufferView' are not supported",
-        );
+        return {
+          // $FlowFixMe[incompatible-cast]
+          data: fromByteArray(new Uint8Array((part))),
+          type: 'string',
+        };
       }
       if (part instanceof Blob) {
         return {
@@ -92,6 +94,7 @@
     }, 0);
 
     NativeBlobModule.createFromParts(items, blobId);
+    // console.error('BlobManager thinks blob id ' + blobId + ' already exists (but it may not yet on iOS!).');
 
     return BlobManager.createFromOptions({
       blobId,
diff --git a/node_modules/react-native-macos/Libraries/Blob/RCTBlobManager.mm b/node_modules/react-native-macos/Libraries/Blob/RCTBlobManager.mm
index 9d4cde7c4..304638bf1 100755
--- a/node_modules/react-native-macos/Libraries/Blob/RCTBlobManager.mm
+++ b/node_modules/react-native-macos/Libraries/Blob/RCTBlobManager.mm
@@ -95,6 +95,7 @@
 - (void)store:(NSData *)data withId:(NSString *)blobId
 {
   std::lock_guard<std::mutex> lock(_blobsMutex);
+  //RCTLogError(@"RCTBlobManager store inside mutex storing blob id %@", blobId);
   _blobs[blobId] = data;
 }
 
@@ -117,9 +118,13 @@
   NSData *data;
   {
     std::lock_guard<std::mutex> lock(_blobsMutex);
+    // FIXME Right here it seems the blob does not exist for some reason?
+    //RCTLogError(@"RCTBlobManager resolve inside mutex resolving blob id %@", blobId);
     data = _blobs[blobId];
   }
   if (!data) {
+    // FIXME and we return nil to handleNetworkingRequest which then crashes
+    //RCTLogError(@"RCTBlobManager resolve no data for blob id %@", blobId);
     return nil;
   }
   if (offset != 0 || (size != -1 && size != data.length)) {
@@ -156,6 +161,7 @@
 - (void)remove:(NSString *)blobId
 {
   std::lock_guard<std::mutex> lock(_blobsMutex);
+  //RCTLogError(@"RCTBlobManager remove inside mutex removing blob id %@", blobId);
   [_blobs removeObjectForKey:blobId];
 }
 
@@ -217,15 +223,19 @@
     }
   }
 
+  //RCTLogError(@"RCTBlobManager createFromParts requesting store of blob id %@", blobId);
   dispatch_async([self executionQueue], ^{
     [self store:data withId:blobId];
+    //RCTLogError(@"RCTBlobManager createFromParts done storing blob id %@", blobId);
   });
 }
 
 RCT_EXPORT_METHOD(release : (NSString *)blobId)
 {
+  //RCTLogError(@"RCTBlobManager enqueueing release of blob id %@", blobId);
   dispatch_async([self executionQueue], ^{
     [self remove:blobId];
+    //RCTLogError(@"RCTBlobManager done releasing blob id %@", blobId);
   });
 }
 
@@ -297,13 +307,30 @@
   // @lint-ignore FBOBJCUNTYPEDCOLLECTION1
   NSDictionary *blob = [RCTConvert NSDictionary:data[@"blob"]];
 
+  // FIXME so here - we checked that data[@"blob"] was non-nil in canHandleHandleNetworkingRequest
+
   NSString *contentType = @"application/octet-stream";
   NSString *blobType = [RCTConvert NSString:RCTNilIfNull(blob[@"type"])];
   if (blobType != nil && blobType.length > 0) {
     contentType = blob[@"type"];
   }
 
-  return @{@"body" : [self resolve:blob], @"contentType" : contentType};
+  // FIXME but here in resolve:blob the blob isn't found somehow? comes back nil?
+  NSData *blobBody = nil;
+  NSString *blobId = nil;
+  for (int retries = 0; blobBody == nil && retries < 100; retries++) {
+    blobBody = [self resolve:blob];
+    if (blobBody == nil) {
+      if (blobId == nil) {
+        blobId = [RCTConvert NSString:blob[@"blobId"]];
+      }
+      //RCTLogError(@"RCTBlobManager handleNetworkingRequest nil blobBody for blob id %@", blobId);
+      //RCTLogError(@"RCTBlobManager handleNetworkingRequest, sleeping for blob id %@", blobId);
+      [NSThread sleepForTimeInterval:0.1f];
+    }
+  }
+
+  return @{@"body" : blobBody, @"contentType" : contentType};
 }
 
 - (BOOL)canHandleNetworkingResponse:(NSString *)responseType
diff --git a/node_modules/react-native-macos/Libraries/Network/RCTNetworking.mm b/node_modules/react-native-macos/Libraries/Network/RCTNetworking.mm
index 80d8c3058..b68a67e77 100644
--- a/node_modules/react-native-macos/Libraries/Network/RCTNetworking.mm
+++ b/node_modules/react-native-macos/Libraries/Network/RCTNetworking.mm
@@ -417,9 +417,19 @@
   for (id<RCTNetworkingRequestHandler> handler in _requestHandlers) {
     if ([handler canHandleNetworkingRequest:query]) {
       // @lint-ignore FBOBJCUNTYPEDCOLLECTION1
-      NSDictionary *body = [handler handleNetworkingRequest:query];
-      if (body) {
-        return callback(nil, body);
+      NSDictionary *blob = [RCTConvert NSDictionary:query[@"blob"]];
+      NSString *blobId = [RCTConvert NSString:blob[@"blobId"]];
+      try {
+        //RCTLogError(@"RCTNetworking process handling blob id %@", blobId);
+        NSDictionary *body = [handler handleNetworkingRequest:query];
+        if (body) {
+          return callback(nil, body);
+        }
+      } catch (NSException *exception) {
+        //RCTLogError(@"RCTNetworking process blob id %@ caught %@ because %@", blobId, exception.name, exception.reason);
+        NSError *error = [[NSError alloc] initWithDomain:NSURLErrorDomain code:NSURLErrorUnknown userInfo:nil];
+        //RCTLogError(@"RCTNetworking process blob id %@ returning error to callback.", blobId);
+        return callback(error, nil);
       }
     }
   }
diff --git a/node_modules/react-native-macos/third-party-podspecs/RCT-Folly.podspec b/node_modules/react-native-macos/third-party-podspecs/RCT-Folly.podspec
index b3c60a50a..d0cd7d498 100644
--- a/node_modules/react-native-macos/third-party-podspecs/RCT-Folly.podspec
+++ b/node_modules/react-native-macos/third-party-podspecs/RCT-Folly.podspec
@@ -24,7 +24,7 @@
   spec.dependency 'DoubleConversion'
   spec.dependency 'glog'
   spec.dependency "fast_float", "6.1.4"
-  spec.dependency "fmt", "11.0.2"
+  spec.dependency "fmt", "12.1.0"
   spec.compiler_flags = folly_compiler_flags + ' -DFOLLY_HAVE_PTHREAD=1 -Wno-documentation -faligned-new'
   spec.source_files = 'folly/String.cpp',
                       'folly/Conv.cpp',
diff --git a/node_modules/react-native-macos/third-party-podspecs/fmt.podspec b/node_modules/react-native-macos/third-party-podspecs/fmt.podspec
index 2f38990e2..a40c5755e 100644
--- a/node_modules/react-native-macos/third-party-podspecs/fmt.podspec
+++ b/node_modules/react-native-macos/third-party-podspecs/fmt.podspec
@@ -8,14 +8,14 @@
 
 Pod::Spec.new do |spec|
   spec.name = "fmt"
-  spec.version = "11.0.2"
+  spec.version = "12.1.0"
   spec.license = { :type => "MIT" }
   spec.homepage = "https://github.com/fmtlib/fmt"
   spec.summary = "{fmt} is an open-source formatting library for C++. It can be used as a safe and fast alternative to (s)printf and iostreams."
   spec.authors = "The fmt contributors"
   spec.source = {
     :git => fmt_git_url,
-    :tag => "11.0.2"
+    :tag => "12.1.0"
   }
   spec.pod_target_xcconfig = {
     "CLANG_CXX_LANGUAGE_STANDARD" => rct_cxx_language_standard(),
diff --git a/node_modules/react-native-macos/React/CoreModules/React-CoreModules.podspec b/node_modules/react-native-macos/React/CoreModules/React-CoreModules.podspec
index ca1136800..71eb7ce9e 100644
--- a/node_modules/react-native-macos/React/CoreModules/React-CoreModules.podspec
+++ b/node_modules/react-native-macos/React/CoreModules/React-CoreModules.podspec
@@ -65,7 +65,7 @@
   # macOS]
   s.dependency "DoubleConversion"
   s.dependency "fast_float", "6.1.4"
-  s.dependency "fmt", "11.0.2"
+  s.dependency "fmt", "12.1.0"
   s.dependency "RCT-Folly", folly_version
   s.dependency "RCTTypeSafety", version
   s.dependency "React-Core/CoreModulesHeaders", version
diff --git a/node_modules/react-native-macos/Libraries/Blob/React-RCTBlob.podspec b/node_modules/react-native-macos/Libraries/Blob/React-RCTBlob.podspec
index 9d758c174..6dfb3ab9d 100644
--- a/node_modules/react-native-macos/Libraries/Blob/React-RCTBlob.podspec
+++ b/node_modules/react-native-macos/Libraries/Blob/React-RCTBlob.podspec
@@ -50,7 +50,7 @@
 
   s.dependency "DoubleConversion"
   s.dependency "fast_float", "6.1.4"
-  s.dependency "fmt", "11.0.2"
+  s.dependency "fmt", "12.1.0"
   s.dependency "RCT-Folly", folly_version
   s.dependency "React-jsi"
   s.dependency "React-Core/RCTBlobHeaders"
diff --git a/node_modules/react-native-macos/ReactCommon/react/renderer/graphics/React-graphics.podspec b/node_modules/react-native-macos/ReactCommon/react/renderer/graphics/React-graphics.podspec
index aa624ac25..77c33c774 100644
--- a/node_modules/react-native-macos/ReactCommon/react/renderer/graphics/React-graphics.podspec
+++ b/node_modules/react-native-macos/ReactCommon/react/renderer/graphics/React-graphics.podspec
@@ -70,7 +70,7 @@
   s.dependency "React-Core" # [macOS] Needed for RCTUIKit
   s.dependency "DoubleConversion"
   s.dependency "fast_float", "6.1.4"
-  s.dependency "fmt", "11.0.2"
+  s.dependency "fmt", "12.1.0"
   
   if ENV["USE_HERMES"] == nil || ENV["USE_HERMES"] == "1"
     s.dependency "hermes-engine"
diff --git a/node_modules/react-native-macos/ReactCommon/react/renderer/debug/React-rendererdebug.podspec b/node_modules/react-native-macos/ReactCommon/react/renderer/debug/React-rendererdebug.podspec
index 52dc39c87..834a33ab6 100644
--- a/node_modules/react-native-macos/ReactCommon/react/renderer/debug/React-rendererdebug.podspec
+++ b/node_modules/react-native-macos/ReactCommon/react/renderer/debug/React-rendererdebug.podspec
@@ -59,6 +59,6 @@
   s.dependency "RCT-Folly", folly_version
   s.dependency "DoubleConversion"
   s.dependency "fast_float", "6.1.4"
-  s.dependency "fmt", "11.0.2"
+  s.dependency "fmt", "12.1.0"
   add_dependency(s, "React-debug")
 end
diff --git a/node_modules/react-native-macos/ReactCommon/react/nativemodule/samples/ReactCommon-Samples.podspec b/node_modules/react-native-macos/ReactCommon/react/nativemodule/samples/ReactCommon-Samples.podspec
index 68b579a7b..229b5baf6 100644
--- a/node_modules/react-native-macos/ReactCommon/react/nativemodule/samples/ReactCommon-Samples.podspec
+++ b/node_modules/react-native-macos/ReactCommon/react/nativemodule/samples/ReactCommon-Samples.podspec
@@ -69,7 +69,7 @@
   s.dependency "RCT-Folly"
   s.dependency "DoubleConversion"
   s.dependency "fast_float", "6.1.4"
-  s.dependency "fmt", "11.0.2"
+  s.dependency "fmt", "12.1.0"
   s.dependency "React-Core"
   s.dependency "React-cxxreact"
   s.dependency "React-jsi"
diff --git a/node_modules/react-native-macos/ReactCommon/hermes/React-hermes.podspec b/node_modules/react-native-macos/ReactCommon/hermes/React-hermes.podspec
index 3a072cf9d..bb87ec8d9 100644
--- a/node_modules/react-native-macos/ReactCommon/hermes/React-hermes.podspec
+++ b/node_modules/react-native-macos/ReactCommon/hermes/React-hermes.podspec
@@ -48,7 +48,7 @@
   s.dependency "RCT-Folly", folly_version
   s.dependency "DoubleConversion"
   s.dependency "fast_float", "6.1.4"
-  s.dependency "fmt", "11.0.2"
+  s.dependency "fmt", "12.1.0"
   s.dependency "glog"
   s.dependency "hermes-engine"
   s.dependency "React-jsi"
diff --git a/node_modules/react-native-macos/ReactCommon/cxxreact/React-cxxreact.podspec b/node_modules/react-native-macos/ReactCommon/cxxreact/React-cxxreact.podspec
index 2f05b75b7..00c51f48f 100644
--- a/node_modules/react-native-macos/ReactCommon/cxxreact/React-cxxreact.podspec
+++ b/node_modules/react-native-macos/ReactCommon/cxxreact/React-cxxreact.podspec
@@ -44,7 +44,7 @@
   s.dependency "boost"
   s.dependency "DoubleConversion"
   s.dependency "fast_float", "6.1.4"
-  s.dependency "fmt", "11.0.2"
+  s.dependency "fmt", "12.1.0"
   s.dependency "RCT-Folly", folly_version
   s.dependency "glog"
   add_dependency(s, "React-jsinspector", :framework_name => 'jsinspector_modern')
diff --git a/node_modules/react-native-macos/ReactCommon/jsi/React-jsi.podspec b/node_modules/react-native-macos/ReactCommon/jsi/React-jsi.podspec
index fb3cbe955..dc84e5da7 100644
--- a/node_modules/react-native-macos/ReactCommon/jsi/React-jsi.podspec
+++ b/node_modules/react-native-macos/ReactCommon/jsi/React-jsi.podspec
@@ -47,7 +47,7 @@
   s.dependency "boost"
   s.dependency "DoubleConversion"
   s.dependency "fast_float", "6.1.4"
-  s.dependency "fmt", "11.0.2"
+  s.dependency "fmt", "12.1.0"
   s.dependency "RCT-Folly", folly_version
   s.dependency "glog"
 
diff --git a/node_modules/react-native-macos/ReactCommon/jsiexecutor/React-jsiexecutor.podspec b/node_modules/react-native-macos/ReactCommon/jsiexecutor/React-jsiexecutor.podspec
index 17641842b..5b1af8066 100644
--- a/node_modules/react-native-macos/ReactCommon/jsiexecutor/React-jsiexecutor.podspec
+++ b/node_modules/react-native-macos/ReactCommon/jsiexecutor/React-jsiexecutor.podspec
@@ -43,7 +43,7 @@
   s.dependency "RCT-Folly", folly_version
   s.dependency "DoubleConversion"
   s.dependency "fast_float", "6.1.4"
-  s.dependency "fmt", "11.0.2"
+  s.dependency "fmt", "12.1.0"
   s.dependency "glog"
   add_dependency(s, "React-jsinspector", :framework_name => 'jsinspector_modern')
 
diff --git a/node_modules/react-native-macos/ReactCommon/React-Fabric.podspec b/node_modules/react-native-macos/ReactCommon/React-Fabric.podspec
index 08b5134be..b0db245c3 100644
--- a/node_modules/react-native-macos/ReactCommon/React-Fabric.podspec
+++ b/node_modules/react-native-macos/ReactCommon/React-Fabric.podspec
@@ -55,7 +55,7 @@
   s.dependency "glog"
   s.dependency "DoubleConversion"
   s.dependency "fast_float", "6.1.4"
-  s.dependency "fmt", "11.0.2"
+  s.dependency "fmt", "12.1.0"
   s.dependency "React-Core"
   s.dependency "React-debug"
   s.dependency "React-featureflags"
diff --git a/node_modules/react-native-macos/ReactCommon/React-FabricComponents.podspec b/node_modules/react-native-macos/ReactCommon/React-FabricComponents.podspec
index 33f4dd2ff..0263c53a8 100644
--- a/node_modules/react-native-macos/ReactCommon/React-FabricComponents.podspec
+++ b/node_modules/react-native-macos/ReactCommon/React-FabricComponents.podspec
@@ -77,7 +77,7 @@
   s.dependency "glog"
   s.dependency "DoubleConversion"
   s.dependency "fast_float", "6.1.4"
-  s.dependency "fmt", "11.0.2"
+  s.dependency "fmt", "12.1.0"
   s.dependency "React-Core"
   s.dependency "React-debug"
   s.dependency "React-featureflags"
diff --git a/node_modules/react-native-macos/ReactCommon/React-FabricImage.podspec b/node_modules/react-native-macos/ReactCommon/React-FabricImage.podspec
index 0fa0a4017..87ad5f8b1 100644
--- a/node_modules/react-native-macos/ReactCommon/React-FabricImage.podspec
+++ b/node_modules/react-native-macos/ReactCommon/React-FabricImage.podspec
@@ -77,7 +77,7 @@
   s.dependency "glog"
   s.dependency "DoubleConversion"
   s.dependency "fast_float", "6.1.4"
-  s.dependency "fmt", "11.0.2"
+  s.dependency "fmt", "12.1.0"
   s.dependency "React-featureflags"
   s.dependency "React-utils"
   s.dependency "Yoga"
diff --git a/node_modules/react-native-macos/ReactCommon/ReactCommon.podspec b/node_modules/react-native-macos/ReactCommon/ReactCommon.podspec
index a6020da9f..080372dfa 100644
--- a/node_modules/react-native-macos/ReactCommon/ReactCommon.podspec
+++ b/node_modules/react-native-macos/ReactCommon/ReactCommon.podspec
@@ -55,7 +55,7 @@
     ss.dependency "React-logger", version
     ss.dependency "DoubleConversion"
     ss.dependency "fast_float", "6.1.4"
-    ss.dependency "fmt", "11.0.2"
+    ss.dependency "fmt", "12.1.0"
     ss.dependency "glog"
     if using_hermes
       ss.dependency "hermes-engine"
